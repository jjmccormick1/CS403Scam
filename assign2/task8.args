    (define t0 (newBST 5))
    (define t1 (insertBST t0 2))
    (define t3 (insertBST t1 8))
    (define t4 (insertBST t3 9))
    (define t5 (insertBST t4 10))
    (define t6 (insertBST t5 11))
    (define t7 (insertBST t6 1))
    (define t8 (insertBST t7 4))
    (displayBST t8)
