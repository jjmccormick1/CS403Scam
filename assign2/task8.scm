(define (main)
        (setPort (open (getElement ScamArgs 1) 'read))
        (define env this)
        (define (iter expr)
            (if (not (eof?)) (begin (eval expr env) (iter (readExpr))))
        )
        (iter (readExpr))
        (env)
)


(define (node value left right)
        (define (display) (print value))
        this
)

(define (newBST value)
        (node value nil nil)
)

(define (displayBST root)
    (define (iter root indent)
            (if (valid? root)
                (begin
                    (iter (root'right) (string+ indent "    "))
                    (print indent)
                    ((root'display))
                    (println)
                    (iter (root'left) (string+ indent "    "))
                    )
                )
            )
    (iter root "")
)

(define (insertBST root val)
    (cond ((null? root) (node val nil nil))
          ((= (root'value) val) root)
          ((< val (root'value))
            (node (root'value) 
                  (insertBST (root'left) val)
                  (root'right)))
          (else (node (root'value)
                      (root'left)
                      (insertBST (root'right) val)))
                      
    )
)
