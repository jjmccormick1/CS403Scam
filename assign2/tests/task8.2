(define t1 (newBST 70))
(define t2 (insertBST t1 58))
(define t3 (insertBST t2 26))
(define t4 (insertBST t3 49))
(define t5 (insertBST t4 50))
(define t6 (insertBST t5 58))
(define t7 (insertBST t6 47))
(define t8 (insertBST t7 43))
(define t9 (insertBST t8 47))
(define t10 (insertBST t9 58))
(define t11 (insertBST t10 82))
(define t12 (insertBST t11 34))
(define t13 (insertBST t12 87))
(displayBST t13)
