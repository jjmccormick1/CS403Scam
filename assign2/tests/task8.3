(define t1 (newBST 52))
(define t2 (insertBST t1 11))
(define t3 (insertBST t2 18))
(define t4 (insertBST t3 66))
(define t5 (insertBST t4 10))
(define t6 (insertBST t5 87))
(define t7 (insertBST t6 44))
(define t8 (insertBST t7 74))
(define t9 (insertBST t8 75))
(define t10 (insertBST t9 14))
(define t11 (insertBST t10 14))
(define t12 (insertBST t11 62))
(define t13 (insertBST t12 93))
(define t14 (insertBST t13 10))
(displayBST t14)
