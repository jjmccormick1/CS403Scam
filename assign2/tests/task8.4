(define t1 (newBST 95))
(define t2 (insertBST t1 34))
(define t3 (insertBST t2 57))
(define t4 (insertBST t3 97))
(define t5 (insertBST t4 27))
(define t6 (insertBST t5 29))
(define t7 (insertBST t6 19))
(define t8 (insertBST t7 37))
(define t9 (insertBST t8 16))
(define t10 (insertBST t9 59))
(define t11 (insertBST t10 23))
(displayBST t11)
