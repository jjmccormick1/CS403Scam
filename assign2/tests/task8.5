(define t1 (newBST 46))
(define t2 (insertBST t1 71))
(define t3 (insertBST t2 29))
(define t4 (insertBST t3 82))
(define t5 (insertBST t4 59))
(define t6 (insertBST t5 10))
(define t7 (insertBST t6 59))
(define t8 (insertBST t7 41))
(define t9 (insertBST t8 25))
(define t10 (insertBST t9 34))
(define t11 (insertBST t10 39))
(displayBST t11)
