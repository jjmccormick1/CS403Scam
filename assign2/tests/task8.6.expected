        86
    72
71
        64
            58
    57
        55
            52
                    34
                30
                    26
